### PR TITLE
Warm donk-pockets once again have omnizine in them

### PR DESCRIPTION
**Pull request by GPeckman**
## About The Pull Request

One of the post-foodening cleanup PRs inadvertently prevented warm donk-pockets from getting the omnizine they were supposed to have. This PR fixes it by adding a new argument to the microwaveable and bakeable elements, which donk-pockets now use. The new argument has also been added to the grillable element for future use.
## Why It's Good For The Game

Bugs aren't good.
## Changelog
:cl:
fix: Warm donk-pockets should now have omnizine in them again.
/:cl:

### CODE REVIEW

File: code/datums/elements/food/microwavable.dm [hunk 1]
```diff
@@ -6,14 +6,18 @@
 	var/atom/default_typepath = /obj/item/food/badrecipe
 	/// Resulting atom typepath on a completed microwave.
 	var/atom/result_typepath
+	/// Reagents that should be added to the result
+	var/list/added_reagents
 
-/datum/element/microwavable/Attach(datum/target, microwave_type)
+/datum/element/microwavable/Attach(datum/target, microwave_type, list/reagents = list())
```
Comment (MrMelbert) [inline code/datums/elements/food/microwavable.dm:12]: Because this is an element, you need to pass a static cached list to it.

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/datums/elements/food/microwavable.dm (hunk 1 above)
Comment (GPeckman) [reply code/datums/elements/food/microwavable.dm:12]: Would changing the declaration for `added_reagents` to `var/list/added_reagents = list()` and then removing the default argument work?

Comment (MrMelbert) [reply code/datums/elements/food/microwavable.dm:12]: No, any place that adds `/datum/element/microwavable` with reagents needs to do `var/static/list/microwave_reagents = list(blah = 1)`.

Comment (MrMelbert) [reply code/datums/elements/food/microwavable.dm:12]: To explain more, bespoke elements know when to re-use an element by checking the arguments you pass into it. 

Lists are objects, so while two identical lists may look identical they have different references, so it makes two "identical" bespoke elements. 

So to get around that we pass static lists into bespoke elements, these static lists are the same object / same list across all types so it passes the exact same list in (and we get the same element as expected).

Comment (GPeckman) [reply code/datums/elements/food/microwavable.dm]: I've made it use static lists now. All the unit tests passed, and all the donks appeared to have the right chems when I tested it in game, so I think I did it right.

File: code/datums/components/grillable.dm [hunk 2]
```diff
@@ -103,6 +106,9 @@
 		BLACKBOX_LOG_FOOD_MADE(grilled_result.type)
 		grilled_result.reagents.clear_reagents()
 		original_object.reagents?.trans_to(grilled_result, original_object.reagents.total_volume)
+		if(added_reagents)
+			for (var/reagent in added_reagents) // Add any new reagents that should be added
+				grilled_result.reagents.add_reagent(reagent, added_reagents[reagent])
```
Comment (Ghommie) [inline code/datums/components/grillable.dm:111]:
```suggestion
			grilled_result.reagents.add_reagent_list(added_reagents)
```

File: code/datums/components/bakeable.dm [hunk 3]
```diff
@@ -70,6 +74,9 @@
 	if(baked_result.reagents && positive_result) //make space and tranfer reagents if it has any & the resulting item isn't bad food or other bad baking result
 		baked_result.reagents.clear_reagents()
 		original_object.reagents.trans_to(baked_result, original_object.reagents.total_volume)
+		if(added_reagents)
+			for (var/reagent in added_reagents) // Add any new reagents that should be added
+				baked_result.reagents.add_reagent(reagent, added_reagents[reagent])
```
Comment (Ghommie) [inline code/datums/components/bakeable.dm:79]:
```suggestion
			baked_result.reagents.add_reagent_list(added_reagents)
```

File: code/datums/elements/food/microwavable.dm [hunk 4]
```diff
@@ -45,6 +49,9 @@
 		BLACKBOX_LOG_FOOD_MADE(result.type)
 		result.reagents.clear_reagents()
 		source.reagents?.trans_to(result, source.reagents.total_volume)
+		if(added_reagents)
+			for (var/reagent in added_reagents) // Add any new reagents that should be added
+				result.reagents.add_reagent(reagent, added_reagents[reagent])
```
Comment (Ghommie) [inline code/datums/elements/food/microwavable.dm:54]:
```suggestion
			result.reagents.add_reagent_list(added_reagents)
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Pull request merged**